modules/claims_api/app/controllers/concerns/claims_api/v2/power_of_attorney_validation.rb

```diff
@@ -40,6 +40,32 @@ def build_dependent_claimant_verification_service(veteran_participant_id:, user_
                                                             poa_code:)
       end
 
+      # the Claimant object is already being validatated below,
+      # so that code has been adjusted and it is being left out of this workflow
+      def validate_address_values
+        address_objects = { veteran: 'veteran', representative: 'representative' }
+
+        address_objects.each do |key, base|
+          address = form_attributes.dig(key.to_s, 'address')
+
+          validate_zip(address, base) if address.present?
+        end
+      end
+
+      def validate_zip(address, base)
+        country = address['country']&.downcase
+        return unless country == 'us'
+
+        zip_code = address['zipCode']
+
+        if zip_code.blank?
+          collect_error_messages(
+            source: "/#{base}/address/zipCode",
+            detail: "If 'countryCode' is 'US' then 'zipCode' is required."
+          )
+        end
+      end
```

Comment:
Looks like we are duplicating logic:

https://github.com/department-of-veterans-affairs/vets-api/blob/751d05609fd423a144f0296f41e9b6c27000871a/modules/claims_api/app/controllers/concerns/claims_api/v2/power_of_attorney_validation.rb#L155-L162

Could we consolidate to one function that takes `address` and `base`, then refactor the claimant address validation to pass in `base: 'claimant'`?

Reply:
Ya good eye on that, updating now